### ISSUE TITLE
Confirm that agents can have a social network as well as be on a network

### ISSUE THREAD

**Issue by jackiekazil** [projectmesa/mesa#514]
The other day we were talking about networks and couldn't remember if we had....

1.  social networks -- where there is one agent per node. The social networks could also be where the agents are nodes, but I was thinking that it might be easier for an agent to be on a node. 
2.  networks as places -- where agents can move from node to node
3. examples that exist for both of these. I am pretty sure examples exist for the first one, but not sure about the second one. 

This might be done, but we are are uncertain if it is and haven't had time. 

Success for this ticket: 
1. Figure out if the agents on networks work this way. If it does .. YAY! If it doesn't -- create a ticket describing what needs to be done. 
2. Looks at tests for this -- does it test for 1 & 2. If it does... Yay! If it doesn't -- create a ticket describing what needs to be done.
3. Examples for both -- do they exist? YAY! If not, then create a ticket that says which ones need to be created to illustrate the functionality.
4. Link tickets appropriately if they are dependent on each other (ie you can't have an example for a functionality that doesn't exist.)

Networks can be found in space.py here: https://github.com/projectmesa/mesa/edit/master/mesa/space.py

Examples with networks we have: 
- https://github.com/projectmesa/mesa/tree/master/examples/boltzmann_wealth_model_network
- https://github.com/projectmesa/mesa/tree/master/examples/virus_on_network

**Comment by joekroese** [comment]
I'm not sure if I've fully understood the issue but I can answer 3. It is possible (and easy!) to represent both situations. In fact, the two example links you have given do one each!

1. **Node=agent**. The 'Virus on Network' example represents a social network, with each agent anchored to exactly one node each and able to spread infection only to its neighbours.
https://github.com/projectmesa/mesa/tree/master/examples/virus_on_network

2. **Node=place**. The 'Boltzmann Wealth Model with Network' represents agents moving from node to node, as if moving places.
https://github.com/projectmesa/mesa/tree/master/examples/boltzmann_wealth_model_network

**Comment by digitaldust** [comment]
may I chime in on this one from a social sciences perspective?

Node=agent: we do a lot of work (survey, interviews) to establish the relations between actors, it will be hard to understand why the agent is "on" a node and not "the" node. The question will be: who's who? who is the node? who is the agent? which represents the actor? why do we need the other one then?

Node=Place: in my field I have not seen an example where networks are used as a representation of places: if space is an issue, then generally a GIS approach is preferred.

To sum up: at least in the social sciences, it's meaningful to have social networks where agents are nodes, and if space is involved we usually adopt GIS. Of course, mixing a GIS world with social networks is allowed.

**Comment by tpike3** [comment]
Please reference multilevel mesa for enhanced agent network functionality
